quodlibet/ext/events/synchronize_to_device.py

```diff
@@ -1079,14 +1081,18 @@ def _remove_empty_dirs(self):
         for root, dirs, __ in os.walk(self.expanded_destination, topdown=False):
             for dirname in dirs:
                 dir_path = os.path.realpath(os.path.join(root, dirname))
-                if not os.listdir(dir_path):
+                files = os.listdir(dir_path)
```

Comment:
Though it was there before, `listdir` shouldn't be necessary if `os.walk` is used; the third tuple items is filenames.

This will also be a lot faster since it avoids scanning twice